### ISSUE TITLE
[a11y] Program filters group label is not announced by screen reader

### ISSUE THREAD

**Issue by melanie-exygy** [civiform/civiform#11423]
**Describe the bug**
The program filters form section on the Applicant Homepage is not announced as a control group by the screen reader.  The group label "What do you want help with?" should be announced.

**To Reproduce**

1. As an admin, edit a program and add some program filters to it.
2. Publish the program changes.
3. Go to the applicant homepage.
4. Turn the screen reader on and have it read through the page.
6. Note that the screen reader jumps straight to read each of the checkbox labels, without reading the group label, which is "What do you want help with?"

**Expected behavior**
The screen reader should say "What do you want help with? - group" before reading the checkboxes.

**Screenshots**

**Desktop (please complete the following information):**

- OS:  iOS
- Browser: Chrome
- Screen Reader:  VoiceOver

**Proposed solution**
The `<legend>` element needs to be the direct child of the `<fieldset>` element.  It will then be announced by the screen reader.
